[Bug]: Without MOUSE_EVENTS fails to build

### What happened?

Building conky with the following disabled, fails to build:

`ARGB DOUBLE_BUFFER IMLIB2 MOUSE_EVENTS XFT XINERAMA LUA_CAIRO LUA_IMLIB2 LUA_RSVG WAYLAND`

This is the error:

```
/wrkdirs/usr/ports/sysutils/conky-awesome/work/conky-1.23.0/src/output/display-x11.cc:459:3: error: unknown type name 'xi_event_type'
  459 |   xi_event_type event_type = ev.xcookie.evtype;
      |   ^
/wrkdirs/usr/ports/sysutils/conky-awesome/work/conky-1.23.0/src/output/display-x11.cc:463:5: error: use of undeclared identifier 'handle_xi_device_change'
  463 |     handle_xi_device_change(device_change);
      |     ^
/wrkdirs/usr/ports/sysutils/conky-awesome/work/conky-1.23.0/src/output/display-x11.cc:468:16: error: use of undeclared identifier 'xi_event_data'
  468 |   auto *data = xi_event_data::read_cookie(display, ev.xcookie.data);
      |                ^
/wrkdirs/usr/ports/sysutils/conky-awesome/work/conky-1.23.0/src/output/display-x11.cc:502:34: error: member reference base type 'MouseEventDebounceMap' (aka 'in
t') is not a structure or union
  502 |   bool already_handled = debounce.count(hash) > 0;
      |                          ~~~~~~~~^~~~~~
/wrkdirs/usr/ports/sysutils/conky-awesome/work/conky-1.23.0/src/output/display-x11.cc:506:28: error: member reference base type 'MouseEventDebounceMap' (aka 'in
t') is not a structure or union
  506 |   for (auto iter = debounce.begin(); iter != debounce.end();) {
      |                    ~~~~~~~~^~~~~~
/wrkdirs/usr/ports/sysutils/conky-awesome/work/conky-1.23.0/src/output/display-x11.cc:506:54: error: member reference base type 'MouseEventDebounceMap' (aka 'int') is not a structure or union
  506 |   for (auto iter = debounce.begin(); iter != debounce.end();) {
      |                                              ~~~~~~~~^~~~
/wrkdirs/usr/ports/sysutils/conky-awesome/work/conky-1.23.0/src/output/display-x11.cc:508:22: error: member reference base type 'MouseEventDebounceMap' (aka 'int') is not a structure or union
  508 |       iter = debounce.erase(iter);
      |              ~~~~~~~~^~~~~~
/wrkdirs/usr/ports/sysutils/conky-awesome/work/conky-1.23.0/src/output/display-x11.cc:842:33: error: no type named 'xi_event_data' in namespace 'conky'
  842 |       delete static_cast<conky::xi_event_data *>(cookie);
      |                          ~~~~~~~^
/wrkdirs/usr/ports/sysutils/conky-awesome/work/conky-1.23.0/src/output/display-x11.cc:842:7: warning: cannot delete expression with pointer-to-'void' type 'void *' [-Wdelete-incomplete]
  842 |       delete static_cast<conky::xi_event_data *>(cookie);
      |       ^                                          ~~~~~~
1 warning and 8 errors generated.
ninja: build stopped: subcommand failed.
===> Compilation failed unexpectedly.
```

The issue is that the include of mouse-events.h is guarded by the `BUILD_MOUSE_EVENTS` option, but then, the cookie handled is not guarded and conditionally-compiled, resulting in an unknown type error.
I can't provide a patch since I don't really know why the relevant change happened.

### Version

1.23.0

### Which OS/distro are you seeing the problem on?

Linux (other)

### Conky config

```Lua

```

### Stack trace

```GDB

```

### Relevant log output

```Shell

```

--- a/src/output/display-x11.cc
+++ b/src/output/display-x11.cc
@@ -44,9 +44,6 @@
 #ifdef BUILD_IMLIB2
 #include "../conky-imlib2.h"
 #endif /* BUILD_IMLIB2 */
-#ifdef BUILD_MOUSE_EVENTS
-#include "../mouse-events.h"
-#endif /* BUILD_MOUSE_EVENTS */
 #include <X11/extensions/XI2.h>
 #include <X11/extensions/XInput2.h>
 #undef COUNT
@@ -69,6 +66,7 @@
 #include "../geometry.h"
 #include "../logging.h"
 #include "../lua/llua.h"
+#include "../mouse-events.h"
 #include "gui.h"
 
 #include "../lua/x11-settings.h"

